### ISSUE TITLE
KeePassX integration

### ISSUE THREAD

**Issue by twpayne** [twpayne/chezmoi#22]
[KeePassX](https://www.keepassx.org/) is a popular password manager. `chezmoi` could use this to store secrets, instead of storing them in `~/.chezmoi.yaml`.

This could possibly be exposed through a template function.

**Comment by twpayne** [comment]
As far as I can tell, KeePassX does have an API or a specification of the database file format.

Closing this issue until at least one of the above becomes available.

**Comment by franek** [comment]
It seems that https://github.com/keepassxreboot/keepassxc provide a CLI. I think we should be able to use [genericSecret.command](https://github.com/twpayne/chezmoi/blob/master/docs/HOWTO.md#use-a-generic-tool-to-keep-your-secrets) to use it. keepassxc-cli can be used like that : ```keepassxc-cli show keepass-file.kdbx passwordkey -a Password```

**Comment by franek** [comment]
With : 
```
"genericSecret.command" = "keepassxc-cli"
``` 
into ~/.config/chezmoi/chezmoi.toml

I can run 
```
chezmoi secret generic -- show ./database.kdbx -a Password id
```
with success.

But, templating seems to not work : 
```
{{ secret "-- show ./database.kdbx -a Password id" }}
```
generate something like that : 
```
[85 115 97 103 46]
```
I don't know why.

**Comment by twpayne** [comment]
I think this is because keepassxc-cli needs to read your password from stdin. I have a branch in progress to add KeePassXC support. Watch this space.

**Comment by twpayne** [comment]
Fixed by #293.

**Comment by franek** [comment]
Many thanks. It is awesome. It works ;)
